src/ComputeSharp.D2D1/Shaders/Interop/D2D1PixelShaderEffect.cs

```diff
@@ -243,26 +363,129 @@ public static unsafe ReadOnlyMemory<byte> GetRegistrationBlob<T>(Func<ID2D1Trans
+    </Property>
+    <Property name='ResourceTextureManager5' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager5'/>
+    </Property>
+    <Property name='ResourceTextureManager6' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager6'/>
+    </Property>
+    <Property name='ResourceTextureManager7' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager7'/>
+    </Property>
+    <Property name='ResourceTextureManager8' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager8'/>
+    </Property>
+    <Property name='ResourceTextureManager9' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager9'/>
+    </Property>
+    <Property name='ResourceTextureManager10' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager10'/>
+    </Property>
+    <Property name='ResourceTextureManager11' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager11'/>
+    </Property>
+    <Property name='ResourceTextureManager12' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager12'/>
+    </Property>
+    <Property name='ResourceTextureManager13' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager13'/>
+    </Property>
+    <Property name='ResourceTextureManager14' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager14'/>
+    </Property>
+    <Property name='ResourceTextureManager15' type='iunknown'>
+        <Property name='DisplayName' type='string' value='ResourceTextureManager15'/>
+    </Property>
 </Effect>");
 
         // Null terminator for the text
         writer.Write((byte)'\0');
 
-        // Property accessors
-#if NET6_0_OR_GREATER
-        nint getBufferPointer = (nint)(delegate* unmanaged<IUnknown*, byte*, uint, uint*, int>)&PixelShaderEffect.GetConstantBuffer;
-        nint setBufferPointer = (nint)(delegate* unmanaged<IUnknown*, byte*, uint, int>)&PixelShaderEffect.SetConstantBuffer;
-#else
-        nint getBufferPointer = Marshal.GetFunctionPointerForDelegate(PixelShaderEffect.GetConstantBufferWrapper);
-        nint setBufferPointer = Marshal.GetFunctionPointerForDelegate(PixelShaderEffect.SetConstantBufferWrapper);
-#endif
-
         // Bindings
-        writer.Write(1);
+        writer.Write(17);
         writer.WriteAsUtf8("Buffer");
         writer.Write((byte)'\0');
-        writer.Write(getBufferPointer);
-        writer.Write(setBufferPointer);
+        writer.Write((nint)PixelShaderEffect.GetConstantBuffer);
+        writer.Write((nint)PixelShaderEffect.SetConstantBuffer);
+        writer.WriteAsUtf8("ResourceTextureManager0");
```

Comment:
Is there something you can grab onto to do a `nameof(ResourceTextureManager0)` with?

Reply:
Yeah we have the constants in `D2D1PixelShaderEffectProperties`, but unfortunately I can't use them here, as all of thise will move to UTF8 literals soon to avoid encoding at runtime, and we can't use `nameof()` with UTF8 literals 😭